app/code/community/PagarMe/CreditCard/Model/Creditcard.php

```
@@ -353,6 +366,13 @@ public function authorize(Varien_Object $payment, $amount)
         } catch (CantCaptureTransaction $exception) {
             Mage::log($exception->getMessage());
             Mage::logException($exception);
+        } catch(ClientException $clientException) {
+            Mage::log('Client exception');
```

When a timeout occurs, shouldn't we refund the transaction via reference key? From what I understand, the change relies in the postback to change the order, but from the client's point of view, woudn't he /she think that the order was completed successfully, when in fact it could be refused?

---

That's a good point o.O

---

If the transaction is refused, a postback will be returned with this stats, is that correctly @henriquekano ?

---

Yep @lflimeira

---

@henriquekano I got you but, the user will get that the order was placed, not approved/captured. After that, the postback will update the order properly notifying the client and the e-commerce about the order status. Makes sense?

---

That's another good point :open_mouth:

---

@leonampd that's exactly the problem. This async pattern is not really seen with good eyes. Unfortunately, from experience, the expectation is something more synchronous: you either get to the end of the order (= EVERYTHING went super duper great) or you don't (= something went wrong, imma fix it right now); even if there's an email sent, some argue it's a missable sign, both from the point of view of the buyer and from the point of view of the seller, which could mean a missed opportunity 
Now, given that, I'd at least add a configuration about if the module should throw an exception and a refund posted in case of a timeout or leave it async.